### PR TITLE
Apparatus trigger list

### PR DESCRIPTION
**Pull request by Ysabz**
### Summary 
This PR adds the action list for each selected apparatus. When the user selects an apparatus from the appratus box , the acion list show the list of available actions. the user can add these actions to the action sequence box by clicking on the + button.
### Details
- The user can select an apparatus compnent and see the available actions [ only void actions]
- The user can click on + next to each action to add it to the sequence box
- the user can remove an action from the sequence box by clicking on the bin 
- 
### How to test?
- run the app
- select an appratus from the selection menu
- you should see the appratus component in the appratus box
- Click on any of the appratuses, if they support an action, you should see them in the action list. [only wobble sphere supports action for now] 
- Click on the + and add as many actions to the action sequence box
- you can remove any number of actions from the action sequence box
### Checks 
- [ X] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

**Comment by JustinLohKingWei** [comment]
Seems you are adding MUI icons in this PR, which requires it to be added to the Node Modules folder via` yarn add @mui/icons-material`. Have you checked that the Git Ignore does not prevent this?

File: src/components/Overlay/Overlay.tsx [hunk 1]
```diff
@@ -136,7 +135,7 @@ function Overlay(): JSX.Element {
           <OverlayGrid>
```
Comment (shushengfuqin) [inline src/components/Overlay/Overlay.tsx:135]: It feels like duplication from lines 112 to 129. This is also the reason why it is not passing sonarqube. Is there a specific reason for this dupliaction? For is there any way to remove the duplication without over-commitment?

**Review by shushengfuqin — COMMENTED** [review]
It is not passing the sonarqube test due to this duplication.

File: src/components/Overlay/Overlay.tsx (hunk 1 above)
Comment (Ysabz) [reply src/components/Overlay/Overlay.tsx]: fixed

Comment (shushengfuqin) [reply src/components/Overlay/Overlay.tsx]: perfect. testing according to your list

**Comment by shushengfuqin** [comment]
I followed your instruction and here are what I see as a result. If they are what they are supposed to do, then it's good. 
Starting page: localhost:3000, I have the choice of evil-cylinder and wobble sphere.

When choosing evil-cylinder,
Screen Shot 2021-11-22 at 20 08 48
there is no trigger for the evil cylinder.  When changing to a wobble sphere under the evil cylinder, the triggers do show up. however, they do not have any responsese. 

When choosing wobble cylinder, 
There are 3 triggers (still, wobble, bounce). Clicking the button itself does add to the added trigger list, however, the object does act accordingly, the plus sign works fine (personal idea, maybe it will be better to click the button instead of the plus sign. or make the plus sign more visible). Buttons in the added trigger list are not responding. 
The delete button does works. 

Here are what I see. If they are perfectly responding to your prediction, then I'm good with it

However, great job!!!

**Review by JustinLohKingWei — DISMISSED** [review]

**Comment by Ysabz** [comment]
@shushengfuqin , regarding your testing, 
the evil cylinder does not have any triggers and that's why there are no actions shown in the action box. 
The plus sign is added so that the user can only view the action without adding it to the sequence.  
the actions are not reponsive for the evil cylinder because in the current deign when clicking on an apparatus you are loading a new appratus so the wobble sphere you see there is not the same one next to the evil cylinder and that's why selection actions does not perform anything on that wobble sphere

**Comment by Ysabz** [comment]
> Seems you are adding MUI icons in this PR, which requires it to be added to the Node Modules folder via` yarn add @mui/icons-material`. Have you checked that the Git Ignore does not prevent this?
> 
> That aside, functionalites work fine and I would approve this merge once it's confirmed that MUI icons is included in this push

The package.json and yarn.lock are updated

**Review by shushengfuqin — APPROVED** [review]
no more questions from me.

**Comment by JustinLohKingWei** [comment]
> > Seems you are adding MUI icons in this PR, which requires it to be added to the Node Modules folder via` yarn add @mui/icons-material`. Have you checked that the Git Ignore does not prevent this?
> > That aside, functionalites work fine and I would approve this merge once it's confirmed that MUI icons is included in this push
> 
> The package.json and yarn.lock are updated

Good. Nothing else to add from me

**Review by JustinLohKingWei — APPROVED** [review]

**Pull request merged**